src/ctap/storage.rs

```diff
@@ -375,37 +382,60 @@ pub fn set_pin(
             key: properties_key,
             value: &pin_properties[..],
         },
-        StoreUpdate::Remove {
-            key: force_change_key,
+        StoreUpdate::Insert {
+            key: key::FORCE_PIN_CHANGE,
+            value: &value,
         },
     ])?)
 }
 
 /// Returns the number of remaining PIN retries.
 pub fn pin_retries(env: &mut impl Env, slot_id: usize) -> Result<u8, Ctap2StatusCode> {
-    let key = check_and_get_key_for_slot(env, slot_id, key::FIRST_PIN_RETRIES, key::PIN_RETRIES)?;
-    match env.store().find(key)? {
+    match env.store().find(key::PIN_RETRIES)? {
         None => Ok(env.customization().max_pin_retries()),
-        Some(value) if value.len() == 1 => Ok(value[0]),
-        _ => Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR),
+        Some(value) => {
+            if slot_id >= value.len() {
+                Ok(env.customization().max_pin_retries())
+            } else {
+                Ok(value[slot_id])
```

Comment:
PIN retries change pretty often actually. Every time you enter a PIN, they are decreased, and then reset after the check has successfully completed. They should be treated like counters and have separate storage keys.